web/components/confirm-password-reset.jsx

```diff
@@ -0,0 +1,137 @@
+import AlertActions from '../actions/alert-actions';
+import Formsy from 'formsy-react';
+import PageHeader from './controls/page-header.jsx';
+import passwordStrengthRegex from '../../service/utils/password-strength-regex';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import request from '../request-agent';
+import Spinner from './controls/spinner.jsx';
+import TextBox from './controls/text-box.jsx';
+import { withRouter } from 'react-router';
+
+import {
+	Button,
+	Col,
+	Grid,
+	Row
+} from 'react-bootstrap';
+
+const ALERT_KEY = 'confirm-password-reset';
+
+class ConfirmPasswordReset extends React.Component {
+	constructor(props) {
+		super();
+		const query = queryString.parse(props.location.search);
+		this.state = {
+			user: query.user,
+			token: query.token,
+			submitting: false
+		};
+		this.submit = this.submit.bind(this);
+	}
+
+	submit(model, reset, invalidate) {
+		AlertActions.dismissAlert(ALERT_KEY);
+		this.setState(
+			Object.assign(
+				{},
+				this.state,
+				{ submitting: true }));
+		request
+			.post(`/api/auth/${this.state.user}/resetPassword`)
+			.send({
+				token: this.state.token,
+				newPassword: model.newPassword
+			})
+			.then(() => {
+				reset();
+				this.props.history.push('/login');
+				AlertActions.showSuccess(
+					'login',
+					'Password Has Been Reset',
+					'Your password has been reset. You may now use your new password to log in.');
+			})
+			.catch(err => {
+				if (err.status === 400) {
+					return invalidate({
+						newPassword: 'New password did not meet strength requirements.'
+					});
+				}
+
+				AlertActions.handleErrorResponse(ALERT_KEY, err);
+			})
+			.finally(() => {
+				this.setState(
+					Object.assign(
+						{},
+						this.state,
+						{ submitting: false }));
+			});
+		reset();
+	}
+
+	render() {
+		if (!this.state.user || !this.state.token) {
+			return <Redirect to="/error/notFound" />;
```

Comment:
We don't need an actual `<Redirect />` component here. We can just render the `<NotFound />` component on the same URL.

Reply:
Yup. Much better.